packages/drivers/driver-base/src/documentDeltaConnection.ts

```diff
@@ -109,9 +105,15 @@ export class DocumentDeltaConnection
             if (!DocumentDeltaConnection.eventsToForward.includes(event)) {
                 throw new Error(`DocumentDeltaConnection: Registering for unknown event: ${event}`);
             }
+            assert(!this.disposed, "register for event on disposed object");
+
             // Register for the event on socket.io
             // "error" is special - we already subscribed to it to modify error object on the fly.
-            if (!this.closed && event !== "error" && this.listeners(event).length === 0) {
+            if (event === "error") {
+                return;
+            }
+            assert((this.listeners(event).length !== 0) === this.trackedListeners.has(event), "mismatch");
```

Comment:
Can we better the registered assert error i.e elaborate "mismatch" a bit?

Reply:
I'm putting big comment above it.